src/main/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactory.java

```diff
@@ -109,8 +109,9 @@ public boolean isHostTransition() {
 
     @Override
     public BuildOptions patch(BuildOptionsView options, EventHandler eventHandler) {
-      if (executionPlatform == null) {
-        // No execution platform is known, so don't change anything.
+      if (executionPlatform == null || options.get(CoreOptions.class).isExec) {
```

Comment:
This isn't safe: it's entirely possible to need to build tools in order to build other tools. How does this lead to the error with action_env?

Reply:
I thought that the error of conflicting actions may be caused by having two copies of the same execution transition. Adding this check fixed the --host_action_env conflicting actions issue, so I created this PR to just to make sure it does not break other tests.

Reply:
If the two configurations were exactly the same, they would have the same hash. Can you check what the difference between the two is? I do not think this change is safe to submit.

You can diff two configurations by using the "bazel config" command, with the same flags, after you've run the command to cause the error. The config subcommand checks for configurations present in skyframe's cache, so anything that invalidates the cache will make this fail.

Reply:
The only difference between the 2 configurations is com.google.devtools.build.lib.analysis.config.CoreOptions.action_env where one config includes the env var passed with host_action_env and the other doesn't.

Reply:
I see that the value of `hostActionEnvironment` isn't being kept in `CoreOptions.getHost`. If you add the line, does that deal with the issue?

```
    // Pass host action environment variables
    host.actionEnvironment = hostActionEnvironment;
    host.hostActionEnvironment = hostActionEnvironment;
```